### PR TITLE
Capture the 404 Requests to File

### PR DESCRIPTION
**Pull request by carmelopreeostudios**

### CODE REVIEW

File: classes/Sensors/System.php [hunk 1]
```diff
@@ -266,4 +293,75 @@ public function WPUpdate($automatic)
...
+    /**
+     * Write a new line on 404 log file
+     */
+    private function WriteLog($data, $ip, $username = '')
+    {
+        $upload_dir = wp_upload_dir();
+        $uploadsDirPath = trailingslashit($upload_dir['basedir']).'404s/';
+        // Check directory
+        if (is_dir($uploadsDirPath) && is_readable($uploadsDirPath) && is_writable($uploadsDirPath)) {
+            $fp = $uploadsDirPath . $ip . $username . '.log';
```
Comment (joseforland) [inline classes/Sensors/System.php]: @carmelopreeostudios start with 0 immediately so there's no need to open 2 files.

File: classes/Sensors/System.php [hunk 2]
```diff
@@ -266,4 +293,75 @@ public function WPUpdate($automatic)
...
+
+            if (!$file = fopen($fp, 'a')) {
+                $i = 0;
+                do {
+                    $fp2 = substr($fp, 0, -4) . '_' . $i . '.log';
+                    if ($file = fopen($fp2, 'a')) {
+                        exit;
+                    }
+                } while ($i > 0);
```
Comment (joseforland) [inline classes/Sensors/System.php]: @carmelopreeostudios do...while can be better; example:
```
$fileOpened = false;
do {
  if ($file = fopen()) {
     $fileOpened = true)
  }
} while (!$fileOpened);
```

File: classes/Sensors/System.php [hunk 3]
```diff
@@ -266,4 +293,75 @@ public function WPUpdate($automatic)
             ));
         }
     }
+
+    /**
+     * Purge log files older than one month
+     */
+    public function LogFilesPruning()
+    {
+        if ($this->plugin->GetGlobalOption('purge-log', 0)) {
```
Comment (joseforland) [inline classes/Sensors/System.php]: @carmelopreeostudios let's make this more specific; something like wp-plugin-purge-404-log

File: classes/Sensors/System.php [hunk 4]
```diff
@@ -299,7 +299,7 @@ public function WPUpdate($automatic)
      */
     public function LogFilesPruning()
     {
-        if ($this->plugin->GetGlobalOption('purge-log', 0)) {
+        if ($this->plugin->GetGlobalOption('purge-404-log', 0)) {
```
Comment (carmelopreeostudios) [inline classes/Sensors/System.php]: @joseforland in the name of the option there is already the prefix so the name is: **wsal-purge-404-log**

File: classes/Sensors/System.php [hunk 5]
```diff
@@ -333,13 +333,15 @@ private function WriteLog($data, $ip, $username = '')
...
                 do {
                     $fp2 = substr($fp, 0, -4) . '_' . $i . '.log';
                     if ($file = fopen($fp2, 'a')) {
-                        exit;
+                        $fileOpened = true;
                     }
-                } while ($i > 0);
+                    $i++;
+                } while (!$fileOpened);
             }
```
Comment (joseforland) [inline classes/Sensors/System.php]: @carmelopreeostudios like this you're still not opening the file that was changed last.  if you have

- IP_USERNAME.log
- IP_USERNAME_1.log
- IP_USERNAME_2.log

you will open the first file if it's available.  We need to open _2.log if it's available and if not, create _3.log

**Pull request merged**